src/core/server/server.ts

```diff
@@ -53,31 +43,36 @@ export class Server {
   public async setup() {
     this.log.debug('setting up server');
 
-    // We shouldn't set up http service in two cases:
-    // 1. If `server.autoListen` is explicitly set to `false`.
-    // 2. When the process is run as dev cluster master in which case cluster manager
-    // will fork a dedicated process where http service will be set up instead.
-    let httpSetup: HttpServiceSetup | undefined;
-    const httpConfig = await this.configService
-      .atPath('server', HttpConfig)
-      .pipe(first())
-      .toPromise();
-    if (!this.env.isDevClusterMaster && httpConfig.autoListen) {
-      httpSetup = await this.http.setup();
-    }
+    const httpSetup = await this.http.setup();
+    this.registerDefaultRoute(httpSetup);
 
     const elasticsearchServiceSetup = await this.elasticsearch.setup();
-
     const pluginsSetup = await this.plugins.setup({
       elasticsearch: elasticsearchServiceSetup,
       http: httpSetup,
     });
 
-    await this.legacy.setup({
+    const coreSetup = {
       elasticsearch: elasticsearchServiceSetup,
       http: httpSetup,
       plugins: pluginsSetup,
-    });
+    };
+
+    this.legacy.setup(coreSetup);
```

Comment:
Though `LegacyService#setup` does not contain async code it is still an async function. I think it should either be changed to a normal function or be awaited here so that future changes to it's implementation are resolved if async code is added.